File: Modules/ServerTypes/Announcement/AnnouncementWizardForm.ts
import { TextAreaEditor, MultipleImageUploadEditor, BooleanEditor, DateEditor, StringEditor, LookupEditor, PrefixedContext } from "@serenity-is/corelib";
import { AnnouncementGeneratedEditor } from "@/HumanResource/Announcement/AnnouncementGenerated/AnnouncementGeneratedEditor";
import { initFormType } from "@serenity-is/corelib/q";

export interface AnnouncementWizardForm {
    AnnouncementContent: TextAreaEditor;
    UploadDocument: MultipleImageUploadEditor;
    Immediate: BooleanEditor;
    Delayed: BooleanEditor;
    AnnouncementDateTime: DateEditor;
    AnnouncementTime: StringEditor;
    Remarks: StringEditor;
    BindToOccupation: BooleanEditor;
    BindToDepartment: BooleanEditor;
    BindToDivision: BooleanEditor;
    BindToJobGrade: BooleanEditor;
    BindToSection: BooleanEditor;
    OccupationListActual: LookupEditor;
    DepartmentListActual: LookupEditor;
    DivisionListActual: LookupEditor;
    JobGradeListActual: LookupEditor;
    SectionListActual: LookupEditor;
    EmployeeRowList: LookupEditor;
    All: BooleanEditor;
    AnnouncementList: AnnouncementGeneratedEditor;
}

export class AnnouncementWizardForm extends PrefixedContext {
    static formKey = 'Announcement.AnnouncementWizard';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AnnouncementWizardForm.init)  {
            AnnouncementWizardForm.init = true;

            var w0 = TextAreaEditor;
            var w1 = MultipleImageUploadEditor;
            var w2 = BooleanEditor;
            var w3 = DateEditor;
            var w4 = StringEditor;
            var w5 = LookupEditor;
            var w6 = AnnouncementGeneratedEditor;

            initFormType(AnnouncementWizardForm, [
                'AnnouncementContent', w0,
                'UploadDocument', w1,
                'Immediate', w2,
                'Delayed', w2,
                'AnnouncementDateTime', w3,
                'AnnouncementTime', w4,
                'Remarks', w4,
                'BindToOccupation', w2,
                'BindToDepartment', w2,
                'BindToDivision', w2,
                'BindToJobGrade', w2,
                'BindToSection', w2,
                'OccupationListActual', w5,
                'DepartmentListActual', w5,
                'DivisionListActual', w5,
                'JobGradeListActual', w5,
                'SectionListActual', w5,
                'EmployeeRowList', w5,
                'All', w2,
                'AnnouncementList', w6
            ]);
        }
    }
}

File: Modules/HumanResource/Announcement/AnnouncementWizard/AnnouncementWizardForm.cs
using Serenity.ComponentModel;
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace HRMSoftware.Announcement.Forms;

[FormScript("Announcement.AnnouncementWizard")]
[BasedOnRow(typeof(AnnouncementWizardRow), CheckNames = true)]
public class AnnouncementWizardForm
{
    [Tab("Announcement Content")]

    [TextAreaEditor(Rows = 5)]
    public string AnnouncementContent { get; set; }
    public string UploadDocument { get; set; }

    [Tab("Announcement Wizard")]
    [Category("Announce Time")]

    [OneThirdWidth]
    public bool Immediate { get; set; }
    [OneThirdWidth]
    public bool Delayed { get; set; }

    [HalfWidth]
    public DateTime AnnouncementDateTime { get; set; }
    [HalfWidth]
    public string AnnouncementTime { get; set; }

    [Category("Remarks")]
    public string Remarks { get; set; }

    [Category("Binds")]
    [QuarterWidth]
    public bool BindToOccupation { get; set; }

    [QuarterWidth]
    public bool BindToDepartment { get; set; }

    [QuarterWidth]
    public bool BindToDivision { get; set; }

    [QuarterWidth]
    public bool BindToJobGrade { get; set; }

    [QuarterWidth]
    public bool BindToSection { get; set; }



    [Category("Filters")]

    [HalfWidth]
    public List<int> OccupationListActual { get; set; }
    [HalfWidth]
    public List<int> DepartmentListActual { get; set; }
    [HalfWidth]
    public List<int> DivisionListActual { get; set; }
    [HalfWidth]
    public List<int> JobGradeListActual { get; set; }
    [HalfWidth]
    public List<int> SectionListActual { get; set; }


    [ThreeQuarterWidth]
    public List<int> EmployeeRowList { get; set; }
    [QuarterWidth]
    public int All { get; set; }



    [Tab("Announcement Generated"), IgnoreName, LabelWidth("0"), AnnouncementGeneratedEditor]
    public List<AnnouncementGeneratedRow> AnnouncementList { get; set; }



}